src/Microsoft.SqlTools.ServiceLayer/LanguageServices/AutoCompleteHelper.cs

```diff
@@ -29,305 +29,394 @@ public static class AutoCompleteHelper
-            "policy",
-            "precision",
-            "predicate",
-            "primary",
-            "print",
-            "prior",
-            "proc",
-            "procedure",
-            "public",
-            "query_store",
-            "quoted_identifier",
-            "raiserror",
-            "range",
-            "raw",
-            "read",
-            "read_committed_snapshot",
-            "read_only",
-            "read_write",
-            "readonly",
-            "readtext",
-            "real",
-            "rebuild",
-            "receive",
-            "reconfigure",
-            "recovery",
-            "recursive",
-            "recursive_triggers",
-            "references",
-            "relative",
-            "remove",
-            "reorganize",
-            "required",
-            "restart",
-            "restore",
-            "restrict",
-            "resume",
-            "return",
-            "returns",
-            "revert",
-            "revoke",
-            "rollback",
-            "rollup",
-            "row",
-            "rowcount",
-            "rowguidcol",
-            "rows",
-            "rule",
-            "sample",
-            "save",
-            "schema",
-            "schemabinding",
-            "scoped",
-            "scroll",
-            "secondary",
-            "security",
-            "select",
-            "send",
-            "sent",
-            "sequence",
-            "server",
-            "session",
-            "set",
-            "sets",
-            "setuser",
-            "simple",
-            "smallint",
-            "smallmoney",
-            "snapshot",
-            "sql",
-            "standard",
-            "start",
-            "started",
-            "state",
-            "statement",
-            "static",
-            "statistics",
-            "statistics_norecompute",
-            "status",
-            "stopped",
-            "sysname",
-            "system",
-            "system_time",
-            "table",
-            "take",
-            "target",
-            "then",
-            "throw",
-            "time",
-            "timestamp",
-            "tinyint",
-            "to",
-            "top",
-            "tran",
-            "transaction",
-            "trigger",
-            "truncate",
-            "try",
-            "tsql",
-            "type",
-            "uncommitted",
-            "union",
-            "unique",
-            "uniqueidentifier",
-            "update",
-            "updatetext",
-            "use",
-            "user",
-            "using",
-            "value",
-            "values",
-            "varchar",
-            "version",
-            "view",
-            "waitfor",
-            "when",
-            "where",
-            "while",
-            "with",
-            "within",
-            "without",
-            "writetext",
-            "xact_abort",
-            "xml",
+        {
+            "abs",

```

Comment:
I think adding built-in SQL functions is fine.  Though I'm not sure how effective it will be.  The way the IntelliSense works is that we first try to use the T-SQL parser to get suggestions.  The parser will only provide schema-based suggestions (unfortunately it doesn't support keyword suggestions).  Only if there are no schema-based suggestions then we return this default keyword list.

I suspect that most the code locations where a user would want to call a function they will likely have schema suggestions available (e.g. in column lists, where clauses, having clauses, etc).  So the default completion list would not be provided.

There are couple exceptions, such as when the editor is not connected or when no tables have been listed in the from clause yet.  But I suspect in the most common use-cases the functions still won't be suggested.  Overall, I'm still fine adding them though in case they help in some cases.

Reply:
I see, the TSQL functions should merge with the schema-based suggestions after a space or comma.